src/engine/routing_algorithms/map_matching.cpp

```diff
@@ -248,8 +263,48 @@ SubMatchingList mapMatching(SearchEngineData<Algorithm> &engine_working_data,
                         current_lengths[s_prime] = network_distance;
                         current_pruned[s_prime] = false;
                         model.breakage[t] = false;
+                        current_segment_id[s_prime] = last_id;
                     }
                 }
+                // Here we can add previous states to the current viterbi if we suppose, that user
+                // can stay
+                double precision = DEFAULT_GPS_PRECISION;
+                if (!trace_gps_precision.empty())
+                    precision = (*trace_gps_precision[t]);
```

Comment:
You can move this evaluation one level up to save on de-referencing the optional GPS trace precision.